lib/extensions/ar_virtual.rb

```diff
@@ -617,10 +617,11 @@ def find_with_associations
       def select(*fields)
         return super if block_given? || fields.empty?
         # support virtual attributes by adding an alias to the sql phrase for the column
+        # it does not add an as() if the column already has an as
         # this code is based upon _select()
         fields.flatten!
         fields.map! do |field|
-          if virtual_attribute?(field) && (arel = klass.arel_attribute(field))
+          if virtual_attribute?(field) && (arel = klass.arel_attribute(field)) && arel.respond_to?(:as)
```

Comment:
Yea, think even this whole redefinition of `select` (and `includes`) is not optimal.

@NickLaMuro Are you suggesting to not worry about the `.as()` calls?